plugins/tableselection/plugin.js

```diff
@@ -811,9 +812,12 @@
 		if ( !selectedCells.length ||
 			// It's single range that does not fully contain table element and is not boundary, e.g. collapsed selection within
 			// cell, part of cell etc.
-			( selectedCells.length === 1 && !rangeContainsTableElement( selection.getRanges()[ 0 ] )  && !boundarySelection ) ||
+			( selectedCells.length === 1 && !rangeContainsTableElement( selection.getRanges()[ 0 ] ) && !boundarySelection ) ||
 			// It's a boundary position but with no table pasted.
-			( boundarySelection && !pastedTable ) ) {
+			( boundarySelection && !pastedTable ) ||
+			// There is some selected content before table (#875).
+			( selectionAncestor && selectionAncestor.$.nodeType == 1 && selectionAncestor.hasClass( 'cke_editable' ) )
```

Comment:
Checking `selectionAncestor.hasClass( 'cke_editable' )` is a bad idea. If you want to check if something is an editbale then you can use  [`equals`](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_editable.html#method-equals) method, for example:
```
editor.editable().equals( nodeToCheckWhichMightBeAnEditable );
```